### ISSUE TITLE
Filtered TRACE KG data

### ISSUE THREAD

**Issue by rimon15** [jun-zeng/ShadeWatcher#5]
Hello,

For the TRACE dataset, I found that creating the KG on the whole dataset filters out most of the edges, and it seems that only events from 1523627788470000000 (Fri Apr 13 2018 13:56:28 GMT+0000 to 1523653792155000000 (Fri Apr 13 2018 21:09:52 GMT+0000) are still available. However, then I am not sure how you split the train/test data for TRACE, as the 13th is one of the browser extension backdoor attacks, so it seems the whole period between these 2 timestamps would not be labeled benign, or did you manually label the edges? Also, you mention 4 attack scenarios for the TRACE dataset, are all of these attack scenarios within this timestamp range? Also, I think I understand the format that the data needs to be in for train/test of the recommendation model (i.e. entity2id is made from one-hot ID from nodefact file, relation2id is one-hot ID from the edgefact files) but I am not sure how you create the entity2id file. Do you look at all of the edges from edgefacts and for each node ID create a list of any nodes they touch (i.e. like an adjacency list)? Or is it something else? I read through the paper but I was still confused on this part. Also, how did you determine the edges used in train2id? Overall, it would be great if you could release the code for converting the KG to the embedding representation for train/test of the recommendation model, as I think that would answer most of my questions, including the limited timestamps in the dataset.

Thanks very much,
Rimon

**Comment by jun-zeng** [comment]
Hi Rimon, 

We manually labeled the edges in the DARPA dataset for training and testing. In particular, in our training set, we use only benign system entity interactions. To generate "malicious" samples, we further perform negative sampling as described in APPENDIX C in the paper. The details of attack scenarios (e.g., the timestamp of malicious behaviors) in the DARPA dataset can be found in Table VI in [Watson](https://jun-zeng.github.io/file/watson_paper.pdf). Based on the edgefact files, we generate entity2id files by applying one-hot encoding to encode the constituent system entities and relations. For your reference, we attach our [raw recommendation results](https://github.com/jun-zeng/ShadeWatcher/files/10121195/gtcache.txt) on the Extension Backdoor attack in the DARAP datset.

Thanks,
Jun
